commands.go

```diff
@@ -788,6 +788,50 @@ func (c cmdable) Set(ctx context.Context, key string, value interface{}, expirat
 	return cmd
 }
 
+// SetArgs provides arguments for the SetWithArgs command.
+//
+// Mode can be `NX` or `XX` or empty.
+//
+// Zero expiration means the key has no expiration time.
+// KeepTTL(-1) expiration is a Redis KEEPTTL option to keep existing TTL.
+//
+// When Get is true, the command returns the old value stored at key, or nil when key did not exist.
+type SetArgs struct {
+	Mode     string
+	ExpireAt time.Duration
+	Get      bool
+}
+
+// SetWithArgs provides a way to call the SET command with SetArgs arguments.
+func (c cmdable) SetWithArgs(ctx context.Context, key string, value interface{}, a *SetArgs) *StatusCmd {
```

Comment:
Let's call this `SetArgs` with we already have such name and it is better to follow one convention.